extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcProvider.java

```diff
@@ -0,0 +1,260 @@
+package io.quarkus.oidc.runtime;
+
+import java.security.Key;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.jboss.logging.Logger;
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.consumer.ErrorCodeValidator;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwx.JsonWebStructure;
+import org.jose4j.keys.resolvers.VerificationKeyResolver;
+import org.jose4j.lang.UnresolvableKeyException;
+
+import io.quarkus.oidc.AuthorizationCodeTokens;
+import io.quarkus.oidc.OIDCException;
+import io.quarkus.oidc.OidcConfigurationMetadata;
+import io.quarkus.oidc.OidcTenantConfig;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+import io.quarkus.security.AuthenticationFailedException;
+import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.smallrye.jwt.algorithm.SignatureAlgorithm;
+import io.smallrye.jwt.util.KeyUtils;
+import io.smallrye.mutiny.Uni;
+import io.smallrye.mutiny.subscription.UniEmitter;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class OidcProvider {
```

Comment:
This class contains a lot of `await` wrapped in emitters.

Either we need to say: ok, it's blocking and then remove all the emitters, or we chain our Uni correctly as I don't see anything that can't be made full non-blocking.